torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -55,6 +55,32 @@ struct TORCH_API WorkerInfo {
   const worker_id_t id_;
 };
 
+// Struct that stores all the metadata needed to retry a given RPC.
+struct TORCH_API RpcRetryInfo {
+  RpcRetryInfo(
+      const WorkerInfo& to,
+      Message&& message,
+      std::shared_ptr<FutureMessage> retryFuture,
+      std::shared_ptr<FutureMessage> originalFuture,
```

Comment:
Do we need to take in the shared_ptr by value? This will bump a refcount every time which is ideally overhead that we'd like to avoid.